vsphere/datadog_checks/vsphere/mor_cache.py

```diff
@@ -97,18 +100,48 @@ def mors_batch(self, key, batch_size):
             for batch in cache.mors_batch('key', 100):
                 for name, mor in batch:
                     # use the Mor object here
+        If max_historical_metrics is specified, the function will also limit
+        the size of the batch so that the integration never makes an API call
+        with more than this given amount of historical metrics.
         """
+        if max_historical_metrics is None:
+            max_historical_metrics = float('inf')
         with self._mor_lock:
-            mors_dict = self._mor.get(key)
+            mors_dict = self._mor.get(key, {})
             if mors_dict is None:
-                yield {}
-
-            mor_names = list(mors_dict)
-            mor_names.sort()
-            total = len(mor_names)
-            for idx in range(0, total, batch_size):
-                names_chunk = mor_names[idx : min(idx + batch_size, total)]
-                yield {name: mors_dict[name] for name in names_chunk}
+                raise StopIteration
```

Comment:
I don't see a case where `mors_dict` can be `None`?

Reply:
`self._mors.get(key, {})` could actually return None (tests or something going wrong) so I wanted a way to exit directly.

I changed that to `self._mors.get(key) or {}` so we can continue execution regardless and the generator won't yield anything